cms/views.py

```diff
@@ -1 +1,210 @@
-# Create your views here.
+import json
+import os.path
+import shutil
+
+from urlparse import urlparse
+
+from django.core.exceptions import ValidationError
+from django.http import HttpResponse
+from django.shortcuts import redirect
+from django.conf import settings
+from django.contrib.auth.decorators import login_required
+from django.views.decorators.csrf import csrf_exempt
+
+from git import Repo
+from elasticgit import EG
+
+from cms import mappings, models
+from cms.management.commands.import_from_git import Command
+
+from unicore.content.models import (
+    Category, Page, Localisation as EGLocalisation)
+
+
+def parse_repo_name(repo_url):
+    pr = urlparse(repo_url)
+    _, _, repo_name_dot_ext = pr.path.rpartition('/')
+    if any([
+            repo_name_dot_ext.endswith('.git'),
+            repo_name_dot_ext.endswith('.json')]):
+        repo_name, _, _ = repo_name_dot_ext.partition('.')
+        return repo_name
+    return repo_name_dot_ext
+
+
+def setup_workspace(repo_path, index_prefix):
```

Comment:
I see there's also a `setup_workspace` method in `eg_resync` command. Can this one be re-used instead?